src/commands/bso/fishingcontest.ts

```diff
@@ -0,0 +1,216 @@
+					.join(', ')}.`
+			);
+		}
+		const currentFishType = getCurrentFishType();
+		const validLocs = getValidLocationsForFishType(currentFishType);
+		if (!validLocs.includes(fishingLocation)) {
+			return msg.channel.send(
+				`This Fishing Location isn't valid for todays catch! These ones are: ${validLocs
+					.map(i => i.name)
+					.join(', ')}`
+			);
+		}
+
+		if (!['Contest rod', "Beginner's tackle box"].every(i => msg.author.hasItemEquippedOrInBank(i))) {
+			return msg.channel.send(
+				'You need to buy a Contest rod and a tackle box to compete in the Fishing contest.'
+			);
+		}
+
+		if (msg.flagArgs.reset) {
+			await prisma.fishingContestCatch.deleteMany({
+				where: {
+					user_id: BigInt(msg.author.id)
+				}
+			});
+			await msg.author.settings.reset(UserSettings.CollectionLogBank);
+			return msg.channel.send('Deleted all your caught fish and reset your CL.');
+		}
+
+		if (msg.flagArgs.debugtime) {
+			let str = 'Rotations\n\n';
+			let counter = {
+				cold: 0,
+				warm: 0,
+				ocean: 0,
+				river: 0,
+				lake: 0
+			};
+			for (let i = 0; i < 50; i++) {
+				let d = new Date(Date.now() + i * (Time.Hour * 24));
+				const rotation = getCurrentFishType(d);
+				counter[rotation.temperature]++;
+				counter[rotation.water]++;
+				str += `${d.toDateString()} ${rotation.temperature} ${rotation.water}\n`;
+			}
+			return msg.channel.send({
+				content: JSON.stringify(counter, null, 2),
+				files: [new MessageAttachment(Buffer.from(str, 'utf-8'), 'text.txt')]
+			});
+		}
+
+		if (msg.flagArgs.xphr) {
+			let str = 'Fishing XP Based On Level at Fishing Contest\n\n\n';
+			for (const lvl of [1, 5, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 110, 120]) {
+				const low = (
+					calculateFishingContestXP({ fishingLevel: lvl, fishSizeCM: 10 }) * GLOBAL_BSO_XP_MULTIPLIER
+				).toLocaleString();
+				const high = (
+					calculateFishingContestXP({ fishingLevel: lvl, fishSizeCM: 180 }) * GLOBAL_BSO_XP_MULTIPLIER
+				).toLocaleString();
+				str += `Level ${lvl}: Roughly Between ${low} and ${high}\n`;
+			}
+			return msg.channel.send({
+				files: [new MessageAttachment(Buffer.from(str, 'utf-8'), 'text.txt')]
+			});
+		}
+
+		if (msg.flagArgs.debug) {
+			let arr = [];
+			for (let i = 0; i < 1000; i++) {
+				arr.push(await catchFishAtLocation({ user: msg.author, location: fishingLocation }));
+			}
+			return msg.channel.send({
+				files: [
+					new MessageAttachment(
+						Buffer.from(
+							arr
+								.sort((a, b) => b.lengthCentimetres - a.lengthCentimetres)
+								.map(i => i.lengthCentimetres)
+								.join('\n'),
+							'utf-8'
+						),
+						'text.txt'
+					)
+				]
+			});
+		}
+
+		let quantity = 1;
+		let duration = Math.floor(quantity * Time.Minute * 1.69);
+		let quantityBoosts = [];
+
+		const tackleBoxes = [
+			"Champion's tackle box",
+			'Professional tackle box',
+			'Standard tackle box',
+			'Basic tackle box'
+		];
+		for (let i = 0; i < tackleBoxes.length; i++) {
```

Comment:
This code has a few problems.

1. It's counting from best => worst,  so you're doing  `num = 0+1` for the best, and `num = 3+1` for the worst
2. You aren't actually doing anything with `num` except updating the boost message. You aren't actually applying that boost anywhere.